pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -396,15 +395,32 @@ func (bi *backendPoolTypeNodeIP) EnsureHostsInPool(service *v1.Service, nodes []
 		changed               bool
 		numOfAdd, numOfDelete int
 		activeNodes           sets.Set[string]
+		err                   error
 	)
 	if bi.useMultipleStandardLoadBalancers() {
-		activeNodes = bi.getActiveNodesByLoadBalancerName(lbName)
+		if !isLocalService(service) {
+			activeNodes = bi.getActiveNodesByLoadBalancerName(lbName)
+		} else {
+			key := strings.ToLower(getServiceName(service))
+			si, found := bi.getLocalServiceInfo(key)
+			if found && !strings.EqualFold(si.lbName, lbName) {
```

Comment:
What would happen if originally cluster was using single LB and a local service was created (so this map does not contain the service) and then cluster is updated to multiple LB (and service may need to be changed to another LB)? Want to make sure it still works.

Reply:
We reconcile all managed LB's backendpools in the outer caller of ensureHostsInPool.

Reply:
It's ok if the map doesn't contain the service here, we just block the case where the lb name in the map is incorrect.